### PR TITLE
New feature: Adds frontend side to allow the users to edit the dataset description

### PR DESCRIPTION
**Pull request by DavidHernandez**
# Description

On the dataset settings page, adds a form field to edit/add a description or annotation guidelines to the dataset. The form has a live preview of the markdown. Both the editor and the markdown renderer are not coupled to the DatasetDescription component so they could be used in any other place if required, for example, when rendering the description somewhere else. I started the VueX ORM implementation but did not finished it as there is no backend for this yet.

While setting up the local environment I noticed that the frontend does not work on recent nodejs versions (tested it on node 18, current LTS) and there is no documentation about this. I only was able to build the frontend with node 14. 

Also, the frontend README is misleading, as it says to use `yarn` which did not work for me, neither on node 18 or 14. It only worked with `npm`.

I'm creating two new issues for this.

**Type of change**

- [ ] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [ ] New cases on DatasetDescription.component.spec.js.
- [ ] New test case on MarkdownEditor.component.spec.js.
- [ ] New test case on MarkdownElement.component.spec.js.

**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: frontend/components/commons/dataset-description/DatasetDescription.component.vue [hunk 1]
```diff
@@ -1,23 +1,69 @@
...
       required: true,
     },
+    isLoading: {
+      type: Boolean,
+      default: () => true,
+    },
+  },
+  components: {
+    MarkdownEditor
   },
```
Comment (keithCuniah) [inline frontend/components/commons/dataset-description/DatasetDescription.component.vue]: in Nuxt, you are not obliged to import the component. So the `components` section can be remove

**Comment by keithCuniah** [comment]
Nice :)

File: frontend/components/commons/dataset-description/DatasetDescription.component.vue (hunk 1 above)
Comment (DavidHernandez) [reply frontend/components/commons/dataset-description/DatasetDescription.component.vue]: Hi @keithCuniah,

I've fixed it. I usually prefer to have explicit dependencies, as I think it makes the code a little bit more clear. Still, it makes sense to remove it to be consistent with the rest of the project.

File: frontend/components/commons/dataset-description/DatasetDescription.component.vue [hunk 2]
```diff
@@ -1,23 +1,65 @@
...
+    ...mapActions({
+      updateDatasetGuidelines: "entities/datasets/updateDatasetGuidelines",
+    }),
+    async updateGuidelines() {
+      console.log('Saving guidelines')
+      try {
+        await this.updateDatasetGuidelines();
+      }
+      catch (e) {
+        //Do nothing. The updateDatasetGuidelines will fail as there is no backend implementation yet
```
Comment (keithCuniah) [inline frontend/components/commons/dataset-description/DatasetDescription.component.vue]: Maybe a toast notification can be shown when saving markdown failed. See `Notification`  in the front app.

Comment (DavidHernandez) [reply frontend/components/commons/dataset-description/DatasetDescription.component.vue]: Added the notification dispatch. Also added a test to verify it works and other improvements in the rest of the tests.

**Comment by DavidHernandez** [comment]
Resolved conflicts from upstream.

**Comment by davidberenstein1957** [comment]
@DavidHernandez Thanks for the lovely PRs! They are very appreciated:)

**Comment by davidberenstein1957** [comment]
https://github.com/argilla-io/argilla/pull/2953

**Pull request closed without merging**